### PR TITLE
Bump version to 0.7.0

### PR DESCRIPTION
**Pull request by irenedea**
Bumps version to 0.7.0 and removes deprecated features.

### Removals
- [x] Remove triton
- [x] Remove prefixlm
- [x] Remove llama attention patch
- [x] Remove z-loss
- [x] Remove text denoising

Regression tests all passed: `llm-foundry-regression-tests-runner-4bPcjd` `llm-foundry-regression-tests-runner-s1KZPH`

### CODE REVIEW

File: .github/workflows/pr-cpu.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
     branches:
     - main
     - release/*
+    - bump_version_v*
```
Comment (irenedea) [inline .github/workflows/pr-cpu.yaml:11]:
```suggestion
```

File: .github/workflows/pr-gpu.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
     branches:
     - main
     - release/**
+    - bump_version_v*
```
Comment (irenedea) [inline .github/workflows/pr-gpu.yaml:11]:
```suggestion
```

File: .github/workflows/smoketest.yaml [hunk 3]
```diff
@@ -8,6 +8,7 @@ on:
     branches:
     - main
     - release/*
+    - bump_version_v*
```
Comment (irenedea) [inline .github/workflows/smoketest.yaml:11]:
```suggestion
```

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ on:
     branches: [main]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [main]
+    branches: [main, bump_version_v*]
```
Comment (irenedea) [inline .github/workflows/codeql-analysis.yml:19]:
```suggestion
    branches: [main]
```

File: .github/workflows/code-quality.yaml [hunk 5]
```diff
@@ -7,6 +7,7 @@ on:
   pull_request:
     branches:
     - main
+    - bump_version_v*
```
Comment (irenedea) [inline .github/workflows/code-quality.yaml:10]:
```suggestion
```

File: tests/models/layers/test_flash_attn.py [hunk 6]
```diff
@@ -362,13 +360,14 @@ def gen_bias():
 
     output_2.sum().backward()
 
-    assert torch.allclose(output_1, output_2)
+    _assert_approx_equal(output_1, output_2)
```
Comment (irenedea) [inline tests/models/layers/test_flash_attn.py:363]: note: had to change this because torch attention and flash attention produced values that were more different to each other than triton and flash

File: tests/models/hf/test_hf_config.py [hunk 7]
```diff
@@ -83,7 +83,8 @@ def test_tie_weights(tie_word_embeddings: bool):
     },
     {
         'attn_config': {
-            'attn_impl': 'triton'
+            'attn_impl': 'flash',
+            'alibi': False,
```
Comment (dakinggg) [inline tests/models/hf/test_hf_config.py:87]: why did you need to add alibi false?

Comment (irenedea) [reply tests/models/hf/test_hf_config.py:87]: it looks like flash2 isn't installed for gpu tests?

Comment (irenedea) [reply tests/models/hf/test_hf_config.py:87]: And alibi isn't supported with flash2

Comment (dakinggg) [reply tests/models/hf/test_hf_config.py:87]: flash2 is installed for gpu tests

Comment (dakinggg) [reply tests/models/hf/test_hf_config.py:87]: and should work fine with alibi

Comment (irenedea) [reply tests/models/hf/test_hf_config.py]: ah whoops this was for cpu tests which don't have flash2 installed

Comment (irenedea) [reply tests/models/hf/test_hf_config.py]: fixed with a patch

**Review by dakinggg — APPROVED** [review]
LGTM, will see if Alex can take a quick skim

File: llmfoundry/data/finetuning/dataloader.py [hunk 8]
```diff
@@ -552,15 +552,6 @@ def _build_collate_fn(
...
                                              clean_up_tokenization_spaces=True))
-                        print(
-                            '\033[92m{}\033[00m\n'.format('CONTEXT:  '),
-                            tokenizer.decode(batch['input_ids'][
-                                j,
-                                torch.logical_and(
-                                    is_subseq, batch['bidirectional_mask'][j] ==
-                                    1)],
-                                             skip_special_tokens=False,
-                                             clean_up_tokenization_spaces=True))
```
Comment (alextrott16) [inline llmfoundry/data/finetuning/dataloader.py:563]: You shouldn't delete these lines. Instead, find a different way to slice out the context. `batch['bidirectional_mask'][j] == 1` was just a convenient way to do that. The context occurs where `attention_mask==1` and `labels==_HF_IGNORE_INDEX`.

File: llmfoundry/data/finetuning/dataloader.py [hunk 9]
```diff
@@ -578,12 +569,6 @@ def _build_collate_fn(
                                                batch['attention_mask'][j] == 1],
                             skip_special_tokens=False,
                             clean_up_tokenization_spaces=True))
-                    print(
-                        '\033[92m{}\033[00m\n'.format('CONTEXT:  '),
-                        tokenizer.decode(batch['input_ids'][
-                            j, batch['bidirectional_mask'][j] == 1],
-                                         skip_special_tokens=False,
-                                         clean_up_tokenization_spaces=True))
```
Comment (alextrott16) [inline llmfoundry/data/finetuning/dataloader.py:586]: See above comment. Don't get rid of this, just do slicing differently.

File: scripts/train/train.py [hunk 10]
```diff
@@ -55,43 +55,10 @@ def validate_config(cfg: DictConfig):
             loaders.append(eval_loader)
     for loader in loaders:
         if loader.name == 'text':
-            if cfg.model.name in ['hf_prefix_lm', 'hf_t5']:
+            if cfg.model.name in ['hf_t5']:
```
Comment (alextrott16) [inline scripts/train/train.py:58]:
```suggestion
            if cfg.model.name == 'hf_t5':
```

**Review by alextrott16 — CHANGES_REQUESTED** [review]
Minor changes requested. Otherwise LGTM.

RIP denoising.
RIP prefix LM.
